### PR TITLE
allow removing lambda logging handler

### PR DESCRIPTION
**Pull request by r0fls**
This will stop the lambda function from logging to cloudwatch, which is kind of pricey. I tried doing it in the LambdaMode `__init__` but then it wipes all loggers (not per Lambda policy).

### CODE REVIEW

File: c7n/mu.py [hunk 1]
```diff
@@ -639,6 +639,10 @@ def name(self):
         return "custodian-%s" % self.policy.name
 
     @property
+    def log(self):
```
Comment (kapilt) [inline c7n/mu.py]: is this used anywhere? if not probably should be removed. self.log in the codebase typically refers to an actual logger instance.

**Comment by r0fls** [comment]
Yep, thanks for the precautionary warning! I'm definitely utilizing the logging for debugging, this is just for potentially once we get things dialed in and are less worried about errors.

**Comment by r0fls** [comment]
@kapilt are we cool to merge this, or is there anything else I should add? Log rotation is a good solution but I think cloudwatch charges per transfer so I think this is a nice to have addition. We could change it from `log` to something else as the key in the `mode` dict. For example, `enable_logging` or `logging`. 

Also, we should probably create an issue to document the existing deployment options, and update this: http://www.capitalone.io/cloud-custodian/docs/overview/deployment.html?highlight=deployment 😂 to me, searching the docs and finding that image was a big let-down. I think it should point to a page that includes methods for both: 
1) how to deploy traditionally, with EC2, on a cron schedule
2) the lambda deployment methods

**Comment by kapilt** [comment]
its good to merge, i just worry that people might use this because things look good, but then something breaks, and they have no context to understand what happened. for us custodian lambdas are quite often good canaries on service issues.

sounds good on docs, thanks again.

**Pull request merged**